EIPS/eip-2026.md

```diff
@@ -0,0 +1,62 @@
+4. It makes is possible to refund `rentbalance` upon self-destruction - when contract is self-destructed, both `balance` and `rentbalance` are returned.
+5. Prepayments on pre-existing accounts are necessary to prevent hoarding of accounts ahead of this change.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+On and after block `H`, every newly created account gets a new field `rentbalance` of type unsigned 256-bit integer.
+On and after block `H`, any operation that leads to the creation of a new account, deducts the amount `ACCOUNT_PREPAYMENT` from `tx.origin`. This amount is added to the `rentbalance` field of the created account.
+On and after block `H`, any operation that modifies an account that does not yet have `rentbalance` field, deducts the amount `ACCOUNT_PREPAYEMENT` from `tx.origin`. This amount is added to the `rentbalance` field of the modified account. This is a anti-hoarding measure.
+ 
+After prepayments are introduced, there can be two reasons for ether to be deducted from `tx.origin`: purchasing and spending gas, and spending gas for prepayments. Gaslimit of a transaction currently plays a role of safety limit, where `gaslimit`  * `gasprice` represents the maximum amount of wei the sender (`tx.origin`) authorises the transaction to deduct from its account.
+After prepayments are introduced, `gaslimit` * `gasprice` will still represent the maximum amount of wei spend, but it will be used for both gas purchases and prepayments, as necessary.
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+Prior to rent prepayments, other alternatives were considered:
+1. Simple raising of the gas cost - discussed in the Motivation section.
+1. In first version of State Rent proposal, there was no notion of extra levy upon account creation. It created a slight usability issue, where newly created contracts with 0 endowment would be evicted in the same block (when rent is introduced). It delays the benefits of the State Rent programme until the actual introduction of rent (in second or third hard-fork).
+2. In the second version of State Rent proposal, there was a notion of lock-up. It is very similar to rent prepayment, with the different that lock-up would not be covering future rent payments.
```

Comment:
Hm, would it make sense making "first version of State Rent proposal" and "second.." links to the blog post (if those are the canonical sources)?

Reply:
Done